pkg/kubectl/cmd/get/humanreadable_flags.go

```diff
@@ -42,13 +42,14 @@ type HumanPrintFlags struct {
 	WithNamespace      bool
 }
 
-// EnsureWithKind sets the provided GroupKind humanreadable value.
-// If the kind received is non-empty, the "showKind" humanreadable
-// printer option is set to true.
-func (f *HumanPrintFlags) EnsureWithKind(kind schema.GroupKind) error {
-	showKind := !kind.Empty()
-
+// SetKind sets the Kind option
+func (f *HumanPrintFlags) SetKind(kind schema.GroupKind) {
 	f.Kind = kind
+}
+
+// EnsureWithKind sets the "Showkind" humanreadable option to true
+func (f *HumanPrintFlags) EnsureWithKind() error {
+	showKind := true
```

Comment:
```go
showKind := false
if f.Kind != nil {
    showKind = !kind.Empty()
}
```

Reply:
`EnsureWithKind` is called before `SetKind` according to @liggitt suggestion.

Current codes works find because the printer will check both the Kind and ShowKind options:
https://github.com/kubernetes/kubernetes/blob/7136cbce50a87755ae2b7b1f92046e47ad9c62d4/pkg/printers/humanreadable.go#L774